packages/core/components/FileDetails/index.tsx

```diff
@@ -49,63 +49,107 @@ export default function FileDetails(props: FileDetails) {
         if (isFileRenderableImage) thumbnailUriPath = fileDetails.path;
     }
 
+    function resizeHandleOnMouseDown(mouseDownEvent: React.MouseEvent<HTMLDivElement>) {
```

Comment:
Maybe I'm misinterpreting, but I think that resizeDetector hook is meant for responding to changes in the size of an element, rather than doing the work required to change its size. 

The only DOM value actually being manipulated here is the `--file-details-width` attribute on 'root', though admittedly there's a good handful of things being read. IDK, I didn't come across any packages that make resizing super easy, and at some level the DOM manipulation has to happen regardless, so I'm cool at the moment with assuming full control over how it works.

Reply:
Ah you're right, I misunderstood the purpose of that hook. The `ResizableBox` component from this one could do the trick, though: https://www.npmjs.com/package/react-resizable

You definitely don't need a library, but using methods like `document.getElementById()` and `rootElement.style.setProperty()` is a bit of a code smell to me. I would personally see if there is a way to pull these off using React features like refs, state, `useEffect`, etc. Here's one example I found:
https://clubmate.fi/resize-any-element#resizable-react-component

Reply:
Actually, I think the first lib I sent might still work. Check out their "Full example (Functional Component)": https://github.com/maslianok/react-resize-detector

Reply:
One more link: https://blog.theodo.com/2020/11/react-resizeable-split-panels/

I've never had to build something like this myself, so interesting to see what the options are!